frontend/src/components/nav/controls-header.tsx

```diff
@@ -141,6 +184,61 @@ function CredentialsActions() {
   )
 }
 
+function EntitiesActions() {
+  const [createEntityDialogOpen, setCreateEntityDialogOpen] = useState(false)
+  const workspaceId = useWorkspaceId()
+  const queryClient = useQueryClient()
+
+  const handleCreateEntity = async (data: {
+    key: string
+    display_name: string
+    description?: string
+    icon?: string
+  }) => {
+    try {
+      await entitiesCreateEntity({
+        workspaceId,
+        requestBody: {
+          key: data.key,
+          display_name: data.display_name,
+          description: data.description,
+          icon: data.icon,
+        },
+      })
+      queryClient.invalidateQueries({ queryKey: ["entities", workspaceId] })
+      toast({
+        title: "Entity created",
+        description: `${data.display_name} has been created successfully.`,
+      })
```

Comment:
prefer to have this as a custom hook that can be reused across various components